code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubUtils.java

```diff
@@ -125,4 +148,58 @@ public String getTeamOrgById(String id) {
         Map<String, String> teamToOrg = (Map<String, String>) jsonData.get("teamToOrg");
         return  teamToOrg.get(id);
     }
+
+    public boolean isAllowed(List<String> idList, Set<ExternalId> externalIds){
+        boolean whiteListed = getWhitelistedUser(idList) != null;
+        boolean hasAccessToAProject = authDao.hasAccessToAnyProject(externalIds, false, null);
+        switch (ACCESS_MODE.get()) {
+            case "restricted":
+                if (whiteListed || hasAccessToAProject) {
+                    break;
+                }
+                throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
+            case "unrestricted":
+                break;
+            default:
+                throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
+        }
+        return true;
+    }
+
+    protected String getWhitelistedUser(List<String> idList) {
+        if (idList == null) {
+            return null;
+        }
+        if (StringUtils.equals(WHITELISTED_USERS.get(), "*")) {
```

Comment:
Null idList takes precedence because if your list is null then you don't have any one to say is white-listed. There for you get no access.